src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Logging/ILoggerFactoryExtensions.cs

```diff
@@ -2,23 +2,31 @@
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.AspNetCore.Razor;
 
 namespace Microsoft.CodeAnalysis.Razor.Logging;
 
 internal static class ILoggerFactoryExtensions
 {
     public static ILogger GetOrCreateLogger<T>(this ILoggerFactory factory)
     {
-        return factory.GetOrCreateLogger(TrimTypeName(typeof(T).FullName));
+        return factory.GetOrCreateLogger(typeof(T));
+    }
+
+    public static ILogger GetOrCreateLogger(this ILoggerFactory factory, Type type)
+    {
+        return factory.GetOrCreateLogger(TrimTypeName(type.FullName.AssumeNotNull()));
     }
 
     private static string TrimTypeName(string name)
     {
-        string trimmedName;
-        _ = TryTrim(name, "Microsoft.VisualStudio.", out trimmedName) ||
-            TryTrim(name, "Microsoft.AspNetCore.Razor.", out trimmedName);
+        if (TryTrim(name, "Microsoft.VisualStudio.", out var trimmedName) ||
+            TryTrim(name, "Microsoft.AspNetCore.Razor.", out trimmedName))
```

Comment:
I know this is pre-existing, but seems like `"Microsoft.CodeAnalysis.Razor" is an obvious one to add to this list